IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -135,4 +197,51 @@ - (MSIDLastRequestTelemetrySerializedItem *)createSerializedItem
     return [[MSIDLastRequestTelemetrySerializedItem alloc] initWithSchemaVersion:[NSNumber numberWithInteger:self.schemaVersion] defaultFields:defaultFields errorInfo:self.errorsInfo platformFields:nil];
 }
 
+#pragma mark - Private: Save To Disk
+
+- (void)saveTelemetryToDisk
+{
+    dispatch_barrier_sync(self.synchronizationQueue, ^{
+        [NSKeyedArchiver archiveRootObject:self toFile:[self filePathToSavedTelemetry]];
+    });
+}
+
+- (id)initFromDecodedObjectWithSchemaVersion:(NSInteger)schemaVersion silentSuccessfulCount:(NSInteger)silentSuccessfulCount errorsInfo:(NSMutableArray<MSIDRequestTelemetryErrorInfo *>*) errorsInfo
+{
+    self = [super init];
+    if (self)
+    {
+        _schemaVersion = schemaVersion;
```

Comment:
should we fail if schema version is not understood?

Reply:
I think this is a good idea. I've update the code such that the hardcoded schemaVersion we had before is now a constant variable, so it can be compared when creating the lastRequestTelemetry from disk. I think we should fall back to initializing a fresh telemetry object if the schema version has changed since the currently saved telemetry has been sent.